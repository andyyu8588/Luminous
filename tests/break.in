// For compiler/VM testing purpose

for(i from 0 to 5 by 1) {
    if(i equals 3) break;
    print(i);
}

a = 0;
while(a < 10) {
    if(a equals 5) {
        print("Broke out of while loop!");
        break;
    }
    print(a);
    a += 1;
}

b = 10;
while(b > 0) {
    c = b;
    while(c > 0) {
        if(c > 3) break;
        print(c);
        c -= 1;
    }
    print(b);
    b -= 1;
}

for(i from 0 to 5 by 1) {
    for(j from i to 5 by 1) {
        if(j > 0) break;
        while(true) {
            print("J is zero");
            for(n from 0 to 4 by 2) {
                print("Cur num is " + n);
                if(n equals 2) break;
            }
            break;
        }
    }
}