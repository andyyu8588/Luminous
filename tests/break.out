0
1
2
0
1
2
3
4
Broke out of while loop!
